app/workers/cypress_viewport_updater/google_analytics_reports.rb

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+
+require 'google/apis/analyticsreporting_v4'
+
+module CypressViewportUpdater
+  class GoogleAnalyticsReports
+    include Google::Apis::AnalyticsreportingV4
+    include Google::Auth
+    include SentryLogging
+
+    JSON_CREDENTIALS = Settings.google_analytics_cvu.to_json
+    SCOPE = 'https://www.googleapis.com/auth/analytics.readonly'
+    VIEW_ID = '176188361'
+
+    def initialize
+      @analytics = AnalyticsReportingService.new
+      @analytics.authorization = ServiceAccountCredentials.make_creds(
+        json_key_io: StringIO.new(JSON_CREDENTIALS),
+        scope: SCOPE
+      )
+      @reports = nil
+    end
+
+    def request_reports
+      request = GetReportsRequest.new(report_requests: [
+                                        user_report_request,
+                                        viewport_report_request
+                                      ])
+      begin
+        @reports = @analytics.batch_get_reports(request).reports
+      rescue => e
+        log_exception_to_sentry(e)
```

Comment:
Are there any cases where we have to explicitly set `@reports` to `nil` when there's an exception, or is this fine as-is?

Reply:
No, we don't have to set `@reports` to `nil`.